compiler/rustc_typeck/src/variance/constraints.rs

```diff
@@ -304,6 +304,10 @@ impl<'a, 'tcx> ConstraintContext<'a, 'tcx> {
                 self.add_constraints_from_invariant_substs(current, substs, variance);
             }
 
+            ty::TyAlias(_, substs) => {
+                self.add_constraints_from_invariant_substs(current, substs, variance);
```

Comment:
I wonder if we can skip this (and/or we need to recurse onto the aliased type), since ty aliases shouldn't participate in variance, though I don't really know much about variance calculation.

Reply:
I'll let it as is for the time being then until we have a better understanding.

Reply:
It will cause subtle bugs if actually reachable. Change to `bug!` instead and only add logic if we actually find code that hits it.